src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -326,6 +330,35 @@ internal virtual NamedTypeSymbol LookupMetadataType(ref MetadataTypeName emitted
             }
 
 Done:
+            if (isTopLevel
+                && scope is not PENamespaceSymbol
+                && (emittedTypeName.ForcedArity == -1 || emittedTypeName.ForcedArity == emittedTypeName.InferredArity)
```

Comment:
>&& (emittedTypeName.ForcedArity == -1 || emittedTypeName.ForcedArity == emittedTypeName.InferredArity) [](http://example.com/codeflow?start=16&length=102)

It would be good to have a test backing this condition.

Reply:
Added test plan item